apicast/src/policy/cors.lua

```diff
@@ -0,0 +1,95 @@
+--- CORS policy
+-- This policy enables CORS (Cross Origin Resource Sharing) request handling.
+-- The policy is configurable. Users can specify the values for the following
+-- headers in the response:
+--   - Access-Control-Allow-Headers
+--   - Access-Control-Allow-Methods
+--   - Access-Control-Allow-Origin
+--   - Access-Control-Allow-Credentials
+-- By default, those headers are set so all the requests are allowed. For
+-- example, if the request contains the 'Origin' header set to 'example.com',
+-- by default, 'Access-Control-Allow-Origin' in the response will be set to
+-- 'example.com' too.
+
+local concat = table.concat
+
+local policy = require('policy')
+local _M = policy.new('CORS Policy')
+
+local new = _M.new
+
+--- Initialize a CORS policy
+-- @tparam[opt] table config
+-- @field[opt] allow_headers Table with the allowed headers (e.g. Content-Type)
+-- @field[opt] allow_methods Table with the allowed methods (GET, POST, etc.)
+-- @field[opt] allow_origin Allowed origins (e.g. 'http://example.com', '*')
+-- @field[opt] allow_credentials Boolean
+function _M.new(config)
+  local self = new()
+  self.config = config or {}
+  return self
+end
+
+local function set_access_control_allow_headers(allow_headers)
+  local value = allow_headers and concat(allow_headers, ',')
```

Comment:
I think it should be possible to assign a table to the header and nginx should take care of the serialization. That could simplify the code a bit.

Reply:
You're right. I didn't know that.
Changed.